examples/gaussian_process/plot_gpr_co2.py

```diff
@@ -33,32 +33,33 @@
 # We will derive a dataset from the Mauna Loa Observatory that collected air
 # samples. We are interested in estimating the concentration of CO2 and
 # extrapolate it for further year. First, we load the original dataset available
-# in OpenML.
+# in OpenML as a pandas dataframe. This will be replaced with Polars
+# once `fetch_openml` adds a native support for it.
 from sklearn.datasets import fetch_openml
 
 co2 = fetch_openml(data_id=41187, as_frame=True)
 co2.frame.head()
 
 # %%
-# First, we process the original dataframe to create a date index and select
-# only the CO2 column.
-import pandas as pd
+# First, we process the original dataframe to create a date column and select
+# it along with the CO2 column.
+import polars as pl
 
-co2_data = co2.frame
-co2_data["date"] = pd.to_datetime(co2_data[["year", "month", "day"]])
-co2_data = co2_data[["date", "co2"]].set_index("date")
+co2_data = pl.DataFrame({col: co2.frame[col].to_numpy() for col in co2.frame.columns})
```

Comment:
why do we need a round trip here? `pl.DataFrame(pd.DataFrame)` doesn't work?

Reply:
Thank you for your review. Converting from/to pandas requires pyarrow as a dependency, and because scikit-learn doesn't have it, the following error will occur when trying to convert a pandas dataframe to Polars:
```python
ModuleNotFoundError: pa.array requires 'pyarrow' module to be installed
```
There is an old related issue converting `pl.to_pandas` https://github.com/pola-rs/polars/issues/3398

Reply:
Opened an issue on the polars side: https://github.com/pola-rs/polars/issues/15845

Reply:
It looks like `pl.from_dataframe` still needs pyarrow for older Polars versions and the checks fail for the minimum Polars version 0.19.12,

Reply:
Since we don't really depend on polars and it's only used in our CI for docs, I'd be in favor of moving the min version to something that supports this.

WDYT @ogrisel @thomasjpfan

Reply:
It seems that `pl.from_dataframe` works without pyarrow starting from Polars version 0.20.4. I can increase the minimum required Polars version from 0.19.12 to 0.20.4 if you prefer this option.

Reply:
I think that'd be okay.

Reply:
> It seems that pl.from_dataframe works without pyarrow starting from Polars version 0.20.4. I can increase the minimum required Polars version from 0.19.12 to 0.20.4 if you prefer this option.

I am okay with this option.